src/app/qgsrelationmanagerdialog.cpp

```diff
@@ -53,6 +53,13 @@ void QgsRelationManagerDialog::addRelation( const QgsRelation &rel )
   if ( ! rel.isValid() )
     return;
 
+  QString referencingFields, referencedFields;
+  for ( int i = 0; i < rel.fieldPairs().count(); i++ )
+  {
+    referencingFields.append( QStringLiteral( "%1%2" ).arg( referencingFields.isEmpty() ? "" : ", " ).arg( rel.fieldPairs().at( i ).referencingField() ) );
```

Comment:
Wouldn't be better to use intermediary QStringlist and join?

Reply:
I used a simpler approach now, considering that 99.5% of the time we'll have only a single pair of fields.